tests/testthat/test_exponential_decay_corr_func.R

```diff
@@ -0,0 +1,38 @@
+test_that(
+  "Test Exponential Decay Corr Func in Stan agrees with R",
+  {
+    model <- compiled_site_inf_model
+    space_model_fxns <- spatial_fxns
+
+    corr_func_params <- list(
```

Comment:
would be nice to run this test multiple times with multiple random `corr_func_params` parameter sets, including different `phi` and `l` values. Definitely worth checking with `l=1`